app/workers/publishing_api_legacy_tags_worker.rb

```diff
@@ -0,0 +1,72 @@
+class PublishingApiLegacyTagsWorker
+  include Sidekiq::Worker
+  API = Services.publishing_api
+
+  def perform(model_id, model_class)
+    model = model_class.constantize.find(model_id)
+    previous_version, taxon_ids = get_taxon_ids(model)
+    parent_ids = taxon_ids.flat_map { |id| expand_parents(id) }
+    legacy_ids = get_legacy_links(taxon_ids + parent_ids)
+
+    links = { policy_areas: [], policies: [], topics: [] }
+    update_topics(links, legacy_ids, model)
+    update_policies(links, legacy_ids, model)
+    update_policy_areas(links, legacy_ids, model)
+
+    params = { previous_version: previous_version, links: links }
+    API.patch_links(model.content_id, params)
```

Comment:
Is the patch_links necessary? Does Whitehall already synchronise with the Publishing Api?

Reply:
So here I'm being consistent with the TaxonLinkPatcher. You're probably right that saving the model would eventually trigger patch_links, but as I haven't actually changed any of the actual model fields it seems more consistent to do it manually.

Reply:
I'm actually not sure if it does This may be necessary - have to dig through the innards of whitehall